pyresample/test/test_utils.py

```diff
@@ -647,6 +647,313 @@ def test_norotation_yaml(self):
+
+        def validate_llnocrs(adef, cfinfo, lat='lat', lon='lon'):
+            self.assertEqual(adef.shape, (19, 37))
+            xc = adef.projection_x_coords
+            yc = adef.projection_y_coords
+            self.assertEqual(xc[0], -180., msg="Wrong x axis (index 0)")
+            self.assertEqual(xc[1], -180. + 10.0, msg="Wrong x axis (index 1)")
+            self.assertEqual(yc[0], -90., msg="Wrong y axis (index 0)")
+            self.assertEqual(yc[1], -90. + 10.0, msg="Wrong y axis (index 1)")
+            self.assertEqual(cfinfo['lon'], lon)
+            self.assertEqual(cf_info['lat'], lat)
+            self.assertEqual(cf_info['type_of_grid_mapping'], 'latitude_longitude')
+            self.assertEqual(cf_info['x']['varname'], 'lon')
+            self.assertEqual(cf_info['x']['first'], -180.)
+            self.assertEqual(cf_info['y']['varname'], 'lat')
+            self.assertEqual(cf_info['y']['first'], -90.)
+
+        cf_file = os.path.join(os.path.dirname(__file__), 'test_files', 'cf_llnocrs.nc')
+
+        # load using a variable=temp
+        adef, cf_info = load_cf_area(cf_file, 'temp', with_cf_info=True)
+        validate_llnocrs(adef, cf_info)
+
+        # load using a variable=None
+        adef, cf_info = load_cf_area(cf_file, with_cf_info=True)
+        validate_llnocrs(adef, cf_info)
+
+        # load from an opened DataArray
+        with xr.open_dataset(cf_file) as xr_handle:
+            adef, cf_info = load_cf_area(xr_handle,  with_cf_info=True)
+        validate_llnocrs(adef, cf_info)
+
+
+class TestLoadCFArea_Private(unittest.TestCase):
+    """ Test the private routines involved in loading an
+        AreaDefinition from netCDF/CF files. """
+
+    def setUp(self):
+        """ Prepare nc_handles """
+        import xarray as xr
+
+        self.nc_handles = {}
+        for k in ('nh10km', 'llwgs84','llnocrs'):
+            cf_file = os.path.join(os.path.dirname(__file__),
+                                   'test_files', 'cf_' + k + '.nc')
+            self.nc_handles[k] = xr.open_dataset(cf_file,)
+
+    def tearDown(self):
+        """ Close nc_handles """
+        for k in self.nc_handles.keys():
+            self.nc_handles[k].close()
+
+    def test_cf_guess_lonlat(self):
+        from pyresample.utils.cf import _guess_cf_lonlat_varname
+
+        # nominal
+        self.assertEqual(_guess_cf_lonlat_varname(self.nc_handles['nh10km'], 'ice_conc', 'lat'), 'lat',)
+        self.assertEqual(_guess_cf_lonlat_varname(self.nc_handles['nh10km'], 'ice_conc', 'lon'), 'lon',)
+        self.assertEqual(_guess_cf_lonlat_varname(self.nc_handles['llwgs84'], 'temp', 'lat'), 'lat')
+        self.assertEqual(_guess_cf_lonlat_varname(self.nc_handles['llwgs84'], 'temp', 'lon'), 'lon')
+        self.assertEqual(_guess_cf_lonlat_varname(self.nc_handles['llnocrs'], 'temp', 'lat'), 'lat')
+        self.assertEqual(_guess_cf_lonlat_varname(self.nc_handles['llnocrs'], 'temp', 'lon'), 'lon')
```

Comment:
what if the longitude variable name is `longitude` ?

Reply:
Then `_guess_cf_lonlat_varname` would return "longitude" (and we would assertEqual against that).